modules/web/src/app/cluster/details/cluster/edit-cluster/component.ts

```diff
@@ -222,12 +226,19 @@ export class EditClusterComponent implements OnInit, OnDestroy {
           this.isKubeLBEnabled = !!(datacenter.spec.kubelb?.enforced || datacenter.spec.kubelb?.enabled);
           this.isKubeLBEnforced = !!datacenter.spec.kubelb?.enforced;
           this.isCSIDriverDisabled = datacenter.spec.disableCsiDriver;
+          this._provider = datacenter.spec.provider;
         })
       )
       .pipe(switchMap(_ => this._datacenterService.seedSettings(this.datacenter.spec.seed)))
       .pipe(takeUntil(this._unsubscribe))
       .subscribe(seedSettings => (this._seedSettings = seedSettings));
 
+    if (this.cluster.spec.cloud?.[this._provider]?.nodePortsAllowedIPRanges?.cidrBlocks?.length) {
```

Comment:
We initialize `this._provider` inside `getDatacenter` subscription and this code block will run before it, hence `this._provider` should be undefined. Are you sure this works as expected?

Reply:
well it is working as expected but yes it should be inside tab pipe